### PR TITLE
Disable http2 until we have upstream bugs fixed

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Disable http2 until we have upstream bugs fixed

## Motivation and Context
We should revert this PR in the future once we
have upstream bugs fixed regarding http2 behavior

## Regression
No

## How Has This Been Tested?
Using mint tests with HTTPS enabled

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#039e46b5-2c48-48b0-95f3-3e35a884f049) |
<a name="039e46b5-2c48-48b0-95f3-3e35a884f049"></a>
### 7711-8a9efe9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30571
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d413ae1e7cda:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 5 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 49 seconds
(4/14) Running aws-sdk-ruby tests ... done in 12 seconds
(5/14) Running awscli tests ... done in 2 minutes and 8 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 2 minutes and 7 seconds
{
  "name": "mc",
  "duration": "109006",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-4656 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-127  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-127/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-127/datafile-0-b`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-127/datafile-1-kB`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-127/datafile-10-MB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-127/datafile-10-kB`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-127/datafile-1-b`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-127/datafile-11-MB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-127/datafile-5-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-127/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-127/datafile-100-kB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-127/datafile-6-MB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-127/datafile-33-kB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-127/datafile-5243880-b`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-127/datafile-65-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-10-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-11-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nTotal: 104.17 MiB, Transferred: 342.02 MiB, Speed: 3.14 MiB/s"
}

Executed 6 out of 14 tests successfully.
```

File: cmd/http/server.go [hunk 1]
```diff
@@ -211,7 +205,7 @@ func NewServer(addrs []string, handler http.Handler, getCert certs.GetCertificat
 			CipherSuites:             defaultCipherSuites,
 			CurvePreferences:         secureCurves,
 			MinVersion:               tls.VersionTLS12,
-			NextProtos:               []string{"h2", "http/1.1"},
+			NextProtos:               []string{"http/1.1", "h2"},
```
Comment (krisis) [inline cmd/http/server.go]: Could you update the commit message that with this change we are making http1.1 the protocol for internal HTTP/RPC communication while the http2 protocol is still available for (external) clients that demand it?

Comment (harshavardhana) [reply cmd/http/server.go]: Done

**Review by krisis — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#be22c4c9-ae4b-4747-a170-34a478641b8a) |
<a name="be22c4c9-ae4b-4747-a170-34a478641b8a"></a>
### 7711-995fd7d/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31909
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 15b8d7421af8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 49 seconds
(8/14) Running minio-dotnet tests ... done in 34 seconds
(9/14) Running minio-go tests ... done in 47 seconds
(10/14) Running minio-java tests ... FAILED in 43 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName, String prefix)",
  "duration": 1022,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-3carr5o, objectName=minio-java-test-373gand, resource=/minio-java-test-3carr5o/minio-java-test-373gand, requestId=15A39BFF0440EDEF, hostId=93c19c25-40de-4dcb-87ba-8c028c319293)\nrequest={method=DELETE, url=http://72.28.97.60:31909/minio-java-test-3carr5o/minio-java-test-373gand?uploadId=4b31685e-d4f8-43d8-a2f1-8c128c458227, headers=Host: 72.28.97.60:31909\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190531T001015Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190531/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-05-30T23-51-45Z\nVary: Origin\nX-Amz-Request-Id: 15A39BFF0440EDEF\nX-Minio-Deployment-Id: 93c19c25-40de-4dcb-87ba-8c028c319293\nX-Xss-Protection: 1; mode=block\nDate: Fri, 31 May 2019 00:10:16 GMT\nTransfer-Encoding: chunked\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.listIncompleteUploads_test2(FunctionalTest.java:1639), FunctionalTest.runTests(FunctionalTest.java:2989), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```

**Pull request merged**